llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1385,6 +1385,12 @@ def match_addos : GICombineRule<
         [{ return Helper.matchAddOverflow(*${root}, ${matchinfo}); }]),
   (apply [{ Helper.applyBuildFn(*${root}, ${matchinfo}); }])>;
 
+def match_subo_no_overflow : GICombineRule<
+  (defs root:$root, build_fn_matchinfo:$matchinfo),
+  (match (wip_match_opcode G_SSUBO, G_USUBO):$root,
```

Comment:
Could you just use `match` instead of `wip_match_opcode`?

Reply:
Firstly, it was copy-pasted from `match_addos` above. Secondly, I believe we can match only one tree/graph per match. I can split it into two combines if you want.

Reply:
I don't like it either. There is no prior-art for matches with two output operands.

Reply:
It's not a big deal, we can try to covert them en-masse in another change.

Reply:
We real/only issue with `wip_match_opcode`is when the docstring says, e.g.  G_FNEG(G_BUILD_VECTOR), and the user is doing `wip_match_opcode` with G_FNEG. It looses precision.